devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -194,205 +180,113 @@ public void startDev() {
+        final Set<AppArtifactKey> projectDependencies = new HashSet<>();
+        addSelfWithLocalDeps(project, builder, new HashSet<>(), projectDependencies, true);
 
-            JavaPluginConvention javaPluginConvention = project.getConvention().findPlugin(JavaPluginConvention.class);
-            if (javaPluginConvention != null) {
-                context.setSourceJavaVersion(javaPluginConvention.getSourceCompatibility().toString());
-                context.setTargetJvmVersion(javaPluginConvention.getTargetCompatibility().toString());
-            }
-
-            if (getCompilerArgs().isEmpty()) {
-                getJavaCompileTask()
-                        .map(compileTask -> compileTask.getOptions().getCompilerArgs())
-                        .ifPresent(context::setCompilerOptions);
-            } else {
-                context.setCompilerOptions(getCompilerArgs());
+        for (AppDependency appDependency : appModel.getFullDeploymentDeps()) {
+            final AppArtifact appArtifact = appDependency.getArtifact();
+            if (!projectDependencies.contains(new AppArtifactKey(appArtifact.getGroupId(), appArtifact.getArtifactId()))) {
+                appArtifact.getPaths().forEach(p -> {
+                    if (Files.exists(p)) {
+                        addToClassPaths(builder, p.toFile());
+                    }
+                });
             }
+        }
 
-            // this is the jar file we will use to launch the dev mode main class
-            final File tempFile = new File(getBuildDir(), extension.finalName() + "-dev.jar");
-            tempFile.delete();
-            tempFile.deleteOnExit();
-
-            context.setDevModeRunnerJarFile(tempFile);
-            modifyDevModeContext(context);
-            try (ZipOutputStream out = new ZipOutputStream(new FileOutputStream(tempFile))) {
-                out.putNextEntry(new ZipEntry("META-INF/"));
-                Manifest manifest = new Manifest();
-                manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
-                manifest.getMainAttributes().put(Attributes.Name.CLASS_PATH, classPathManifest.toString());
-                manifest.getMainAttributes().put(Attributes.Name.MAIN_CLASS, DevModeMain.class.getName());
-                out.putNextEntry(new ZipEntry("META-INF/MANIFEST.MF"));
-                manifest.write(out);
-
-                out.putNextEntry(new ZipEntry(DevModeMain.DEV_MODE_CONTEXT));
-                ByteArrayOutputStream bytes = new ByteArrayOutputStream();
-                try (ObjectOutputStream obj = new ObjectOutputStream(new DataOutputStream(bytes))) {
-                    obj.writeObject(context);
-                }
-                out.write(bytes.toByteArray());
+        //we also want to add the maven plugin jar to the class path
+        //this allows us to just directly use classes, without messing around copying them
```

Comment:
Should be Gradle here no?

Reply:
Ah, right, thanks. It's apparently been there before, I'll fix it.